### ISSUE TITLE
Load MersenneTwister fails

### ISSUE THREAD

**Issue by stakaz** [JuliaIO/JLD2.jl#112]
Hello, I have posted the issue here because I am not sure whether it is a JLD2 issue or not.

https://discourse.julialang.org/t/jld2-load-of-mersennetwister-does-not-work/14939

**Comment by jingpengw** [comment]
dictionary IO is not working.

**Comment by stakaz** [comment]
Could you explain a little more precise and will this be possible after this will work? (But with older saved jld2 files?). 

Thanks.

**Comment by phereford** [comment]
@stakaz I was able to resolve this by adding the following code on julia 1.1
```
using Random, Random.DSFMT
```

Good luck!

**Comment by stakaz** [comment]
Thank you. I am on julia 1.1.0 but for the whole thing works now even without `Random.DSFMT`. I am closing this issue.

**Comment by marius311** [comment]
For bookkeeping/definiteness, I believe this Issue should be reopened. I cleaned up the MWE a bit, and basically the following gives a reconstruction warning on all versions of JLD2 back to v.0.1.0 (I checked with Julia 1.2). 

```julia
using Random
using JLD2

rng = Random.MersenneTwister(1)
@save "rng.jld2" rng
@load "rng.jld2"
```

Also, reverting https://github.com/JuliaIO/JLD2.jl/pull/141/commits/a6c070bde24417d343d31180d48453c42fddd7e1 from master does not make a difference, the above code still gives a warning. However, before https://github.com/JuliaIO/JLD2.jl/pull/141/commits/a6c070bde24417d343d31180d48453c42fddd7e1, adding `using Random.DSFMT` used to be a workaround, however, afterwards, even that no longer works. Hope this might help someone that knows more about JLD2 than I do to track down the bug.
